rewrite(docs): structure/layout + improved docs (#80)
spicetify/spicetify-docs: docs/getting-started/customization/marketplace.md

---
title: Marketplace
description: 📶 Inject addons from within Spotify!
---

Marketplace is a custom app that offers a dedicated page within Spotify, allowing users to peruse and apply add-ons created by the community. The application facilitates the installation of themes, extensions, and snippets, with the exception of custom apps.

## Why use it?

<details>
<summary>Positives</summary>

1. Huge repository of addons.

> 2. Installation is hassle free and quick.

Reviewer: ```suggestion
2. Installation is quick and hassle-free.
```

> 3. Perfect for the average user who doesnt have time to waste in terminal.

Reviewer: ```suggestion
3. Perfect for the average user who doesn't have time to waste in the terminal.
```

> 4. Color.ini editing inapp.

Reviewer: ```suggestion
4. Change colour schemes in-app.
```

5. Since addons are fetched remotely, they will always be up to date.

6. Reduces user error.

</details>

<details>
<summary>Negatives</summary>

> 1. Fresh installs of Spotify / incorrect updating can cause marketplace to forget your installed addons.

Reviewer: ```suggestion
1. Fresh installs of Spotify / incorrect updating can cause Marketplace to forget your installed addons.
```

2. Marketplace requires internet connection to apply the addons.

> 3. Inability to modify addons source code.

Reviewer: ```suggestion
3. Inability to modify addons' source code.
```

4. Inability to see what addons do at a code level before installing.

> 5. Relies on an active connection to github (and jsdelivr for extensions).

Reviewer: ```suggestion
5. Relies on an active connection to GitHub (and jsdelivr for extensions).
```

</details>

## Installation

### Windows
#### Powershell
```powershell
Invoke-WebRequest -UseBasicParsing "https://raw.githubusercontent.com/spicetify/spicetify-marketplace/main/resources/install.ps1" | Invoke-Expression
```

> ### MacOS/Linux

Reviewer: Make order/capitalization consistent across docs. Idc which comes first. 
```suggestion
### Linux/macOS
```

#### Shell
```bash
curl -fsSL https://raw.githubusercontent.com/spicetify/spicetify-marketplace/main/resources/install.sh | sh
```

## Reset Marketplace {#reset}
### Command
```
Marketplace.Reset(theme, extensions, snippets) -
    Description: Resets localstorage objects for specific addons.
    Parameters:
        - none: Resets all localstorage items.
        - theme (optional): Specifies the theme to reset.
        - extensions (optional): Specifies the extensions to reset.

>         - snippets (optional): Specifies the snippets to reset.

Reviewer: Rewrite this to better outline use. I think you call it like `Marketplace.Reset("extensions", "snippets")` etc? 

From Utils.ts
```ts
export const resetMarketplace = (...categories: ("extensions" | "snippets" | "theme")[]) => { 
// ...
}
```

Author: ? could you elaborate

Reviewer: The wording here makes it seem like you pass in specific extensions that you want to disable. It isn't that granular, and only lets you disable entire categories of things at a time.

Author: ah i see the confusion, i understand thats not how it works ofc but the wording in the descriptions make it seem that way, i was attempting to make it simple enough for the average user to understand hence not just copy pasting the function definition from utils.ts

Reviewer: Added suggestion

> Marketplace.Reset(theme, extensions, snippets) -
>     Description: Resets localstorage objects for specific addons.
>     Parameters:
>         - none: Resets all localstorage items.
>         - theme (optional): Specifies the theme to reset.
>         - extensions (optional): Specifies the extensions to reset.
>         - snippets (optional): Specifies the snippets to reset.

Reviewer: How's this? (Reset is lowercase btw)
```suggestion
Marketplace.reset(theme, extensions, snippets) -
    Description: Resets localstorage objects for specific addons types.
    Parameters:
        - none: Resets all Marketplace localstorage items.
        - theme (optional): Reset the current theme installed by Marketplace.
        - extensions (optional): Reset all extensions installed by Marketplace. 
        - snippets (optional): Reset all snippets.
    Usage: `Marketplace.reset("theme", "extensions")` or `Marketplace.reset()`
```

## Reset Marketplace {#reset}
### Command
```
Marketplace.reset(theme, extensions, snippets) -

>     Description: Resets localstorage objects for specific addons types.

Reviewer: ```suggestion
    Description: Resets localstorage objects for specific addon types.
```
